src/components/map/map-display.vue

```diff
@@ -83,15 +80,17 @@ export default {
         boundsUpdated(bounds) {
             this.bounds = bounds;
         },
-        findCountry(name) {
-            return (
-                this.layer.geojson.features.find(
-                    f => f.properties.ISO_A3 === name
-                ) || null
-            );
+        findCountries() {
+            return ({
+                ...this.layer.geojson,
+                features: this.layer.geojson.features.filter(f => {
+                    return this.selectedCountries.map(c => c.code).includes(f.properties.ISO_A3)
```

Comment:
Good point. And yes it does error when features is empty but I'm not sure which bit is throwing the error so I will look into it.

Reply:
you could try applying a `v-if="selectedCountries.length > 0"` to the `<l-geo-json>` component

Reply:
Where do you think the best place to set the `selectedCountryCodes` would be? In the watch when `selectedCountries` has changed or somewhere else?

Reply:
i would add it to the `computed` properties object, underneath `highlightedCountry`. you don't need a `watch` because computed properties will get refreshed every time the change detection gets called on any of its dependencies. the other benefit is computed properties get cached, but a `watch` is just a function that gets called.

Reply:
Ok I wasn't quite sure how the `computed` bit worked so I'll try it.